crates/polars-plan/src/logical_plan/file_scan.rs

```diff
@@ -43,7 +49,29 @@ impl PartialEq for FileScan {
                 },
             ) => opt_l == opt_r && c_l == c_r,
             #[cfg(feature = "ipc")]
-            (FileScan::Ipc { options: l }, FileScan::Ipc { options: r }) => l == r,
+            (
+                FileScan::Ipc {
+                    options: l,
+                    #[cfg(feature = "cloud")]
+                        cloud_options: c_l,
+                    ..
+                },
+                FileScan::Ipc {
+                    options: r,
+                    #[cfg(feature = "cloud")]
+                        cloud_options: c_r,
+                    ..
+                },
+            ) => {
+                #[cfg(not(feature = "cloud"))]
+                {
+                    l == r
+                }
+                #[cfg(feature = "cloud")]
+                {
+                    l == r && c_l == c_r
+                }
```

Comment:
Wondering if this can be written more succinctly.

Reply:
Let's always add the cloud fields even if not activated. The amount of wasted struct bytes is not worth it as you typically only allocate a few.

In this case the code gets too iffy to justify the feature flag IMO.

Reply:
We could also set the type to `()` when `[cfg(not(feature = "cloud"))]`. What do you think?

Reply:
That's a good one! :+1:

Reply:
Followup PR.

```
#[cfg(not(cloud))]
type Foo = ()


```